### PR TITLE
add some documentation on module design

### PR DESCRIPTION
**Pull request by seanbreckenridge**
hope nothing is too far off -- referenced a lot of the issues we've been talking in for the past few months

hopefully nothing broken on the org mode side -- I can now edit org mode as fast as markdown, did take me a while to figure out how links worked though...

now I just need to learn all the keybindings so I'm actually getting benefit out of org

### CODE REVIEW

**Comment by seanbreckenridge** [comment]
Hmm

Seems some of my links dont work, not sure how github renders these...

https://github.com/seanbreckenridge/HPI-fork/blob/doc-updates/doc/SETUP.org#L467

`See [[file:MODULE_DESIGN.org::#addingmodules][MODULE_DESIGN/adding modules]] for more information`

Thought this was how you did it... perhaps I have to follow the markdown `id`-like structure and link to a slugified version like `#adding-new-modules`?

```
* Adding new modules
:PROPERTIES:
:CUSTOM_ID: addingmodules
:END:
```

**Comment by seanbreckenridge** [comment]
Ah - well. the latest commit just removed the link altogether... o_o

**Comment by karlicoss** [comment]
I think the ::* links definitely won't work properly on Github because they are using header search -- so they are inherently dynamic.

So the ::# syntax is the right way in org-mode, and used to work on Github, but got broken at some point :( https://github.com/novoid/github-orgmode-tests/commit/952a8cb973084d19887e90f02bc2827ebdc928d8 

Also did another check now and it seems that `[[file:MODULE_DESIGN.org#addingmodules][MODULE_DESIGN/adding modules]] ` works on github! But org-mode doesn't understand this and assumes # is just a part of a filename. Ugh. But maybe it's a fair compromise in the meantime considering people would likely be reading github docs (rather than browsing them in emacs)?

Also `CUSTOM_ID` / `ID` properties don't seem to work either -- it seems that haven't for a while considering I have this hack in readme https://github.com/karlicoss/HPI/blame/master/README.org#L279 

Opened PR with these findings so at least I remember it next time: https://github.com/novoid/github-orgmode-tests/pull/4/files

Also seems related to https://github.com/alphapapa/org-make-toc#known-issues

**Comment by seanbreckenridge** [comment]
Ok, that seems to work, at least in githubs rendering. left a comment to this PR there

Link at the bottom of

https://github.com/seanbreckenridge/HPI-fork/blob/doc-updates/doc/SETUP.org#addingmodifying-modules

correctly links

**Comment by seanbreckenridge** [comment]
lots of small changes T__T

hard to know whats going to work -- is almost easier just editing in in the browser so I get the live preview

File: doc/SETUP.org [hunk 1]
```diff
@@ -196,7 +200,7 @@ If you experience issues, feel free to report, but please attach your:
 - OS version
 - python version: =python3 --version=
 - HPI version: =pip3 show HPI=
-- if you see some exception, attach a full log (just make suer there is not private information in it)
+- if you see some exception, attach a full log (just make user there is not private information in it)
```
Comment (karlicoss) [inline doc/SETUP.org:203]: this it was meant to be 'sure' :sweat_smile:

File: doc/SETUP.org [hunk 2]
```diff
@@ -123,6 +126,7 @@ If you're not planning to use private configuration (some modules don't need it)
 
 The configuration usually contains paths to the data on your disk, and some modules have extra settings.
 The config is simply a *python package* (named =my.config=), expected to be in =~/.config/my=.
+If you'd like to change the location of the =my.config= directory, you can set the =MY_CONFIG= environment variable. e.g. in your .bashrc add: ~export MY_CONFIG="$HOME/.my/"~
```
Comment (karlicoss) [inline doc/SETUP.org:129]: github rendering seems messed up here... IIRC it doesn't like quotes inside ~ ~ :(

Comment (seanbreckenridge) [reply doc/SETUP.org:129]: ah

suppose typically `MY_CONFIG=$HOME/.my` would work anyways, as long as your username doesnt have spaces in it...

**Comment by karlicoss** [comment]
> hard to know whats going to work -- is almost easier just editing in in the browser so I get the live preview

yep, it's definitely easier, I do that as well

---------

The doc looks great, thanks for assembling!

File: doc/SETUP.org (hunk 2 above)
Comment (karlicoss) [reply doc/SETUP.org]: Yeah... seems that you're asking for trouble if you do that xD 
Seems that it's prohibited in POSIX thankfully https://unix.stackexchange.com/a/401782/180307

**Comment by seanbreckenridge** [comment]
Alright, finished going over everything again just to make sure there were no broken links/indentation

Probably missed something, but this is probably good enough to merge

**Pull request merged**